src/services/atomic-swap.service.js

```diff
@@ -50,10 +52,70 @@
     throw new Error('Unable to create the proposal on the Atomic Swap Service');
   }
 
+  await addListenedProposal(walletId, id, password);
   return { proposalId: id };
 };
 
+/**
+ * Adds a new proposal to the `listenedProposals` map of a wallet.
+ * @param {string} walletId
+ * @param {string} proposalId
+ * @param {string} password
+ * @return {Promise<void>}
+ */
+const addListenedProposal = async (walletId, proposalId, password) => {
+  // Checking if this wallet map has been initialized already
+  if (!walletListenedProposals.has(walletId)) {
+    walletListenedProposals.set(walletId, new Map());
+    // Will also open the websocket channel
+  }
+
+  const listenedProposals = walletListenedProposals.get(walletId);
+  listenedProposals.set(proposalId, { id: proposalId, password });
+  // Will also add the proposalId to the Atomic Swap Service websocket channel
+};
+
+/**
+ * Removes a proposal from the `listenedProposals` map of a wallet
+ * @param {string} walletId
+ * @param {string} proposalId
+ * @return {Promise<void>}
+ */
+const removeListenedProposal = async (walletId, proposalId) => {
+  // If this wallet map has not been initialized, just ignore the request
+  if (!walletListenedProposals.has(walletId)) {
+    return;
+  }
+
+  const listenedProposals = walletListenedProposals.get(walletId);
+  listenedProposals.delete(proposalId);
```

Comment:
This alert is not relevant for this case, since we're not making queries to a NoSQL database, but only deleting a single non-critical entry on a memory map.